csrc/multidevice/utils.cpp

```diff
@@ -13,22 +13,93 @@
 #include <multidevice/lower_communication.h>
 #include <multidevice/utils.h>
 #include <ops/all_ops.h>
+#include <root_domain_map.h>
 #include <scheduler/utils.h>
 
 #include <c10/util/irange.h>
 
 namespace nvfuser {
 namespace {
 
-std::vector<IterDomain*> getShardedIterDomains(TensorView* tv) {
-  std::vector<IterDomain*> sharded_ids;
+const std::unordered_set<IterDomain*> getShardedIterDomains(TensorView* tv) {
```

Comment:
```suggestion
std::unordered_set<IterDomain*> getShardedIterDomains(TensorView* tv) {
```

I'll try to find some reference, but IIRC marking the return object (not ref) as const doesn't provide practical benefits.

Reply:
https://stackoverflow.com/questions/8716330/purpose-of-returning-by-const-value